dandi/cli/cmd_service_scripts.py

```diff
@@ -123,6 +126,145 @@ def reextract_metadata(url: str, diff: bool, when: str) -> None:
                 asset.set_raw_metadata(mddict)
 
 
+# TODO REMOVE JUST REFERENCE
+# -------------------DANDI-ARCHIVE ---------------------------------
+# def _generate_doi_data(version: Version):
+#     from dandischema.datacite import to_datacite
+#
+#     publish = settings.DANDI_DOI_PUBLISH
+#     # Use the DANDI test datacite instance as a placeholder if PREFIX isn't set
+#     prefix = settings.DANDI_DOI_API_PREFIX or "10.80507"
+#     dandiset_id = version.dandiset.identifier
+#     version_id = version.version
+#     doi = f"{prefix}/dandi.{dandiset_id}/{version_id}"
+#     metadata = version.metadata
+#     metadata["doi"] = doi
+#     return (doi, to_datacite(metadata, publish=publish))
+#
+#
+# def create_doi(version: Version) -> str:
+#     doi, request_body = _generate_doi_data(version)
+#     # If DOI isn't configured, skip the API call
+#     if doi_configured():
+#         try:
+#             requests.post(
+#                 settings.DANDI_DOI_API_URL,
+#                 json=request_body,
+#                 auth=requests.auth.HTTPBasicAuth(
+#                     settings.DANDI_DOI_API_USER,
+#                     settings.DANDI_DOI_API_PASSWORD,
+#                 ),
+#                 timeout=30,
+#             ).raise_for_status()
+#         except requests.exceptions.HTTPError as e:
+#             logger.exception("Failed to create DOI %s", doi)
+#             logger.exception(request_body)
+#             if e.response:
+#                 logger.exception(e.response.text)
+#             raise
+#     return doi
+#
+#
+# -------------------DANDI-ARCHIVE ---------------------------------
+
+
+@service_scripts.command()
+@instance_option()
+@click.option(
+    "-d",
+    "--dandiset",
+    metavar="DANDISET_ID",
+    required=True,
+    help="ID of Dandiset to operate on",
+)
+@click.option(
+    "--dandiset-version",
+    metavar="DANDISET_VERSION_ID",
+    required=True,
+    help="ID of Dandiset Version to operate on",
+)
+def publish_dandiset_version_doi(dandi_instance, dandiset, dandiset_version) -> None:
+    with DandiAPIClient.for_dandi_instance(dandi_instance, authenticate=True) as client:
+        ds = client.get_dandiset(dandiset, dandiset_version, lazy=False)
```

Comment:
TODO handle error

Reply:
We opted not to handle these errors. Since its an admin action, we can just let the errors raise to the user